src/search.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include "move.hpp"
+#include "position.hpp"
+#include "uci.hpp"
+#include "util/types.hpp"
+
+namespace Clockwork {
+
+namespace Search {
+struct Stack {
```

Comment:
Would it be better to use a more safer approach to Search Stack Management?

See: https://github.com/TheBlackPlague/StockDory/blob/9f95dea6f73a5e790d33c5489fa1266dbf5e03d7/src/Engine/Search.h#L77-L98

One can index on `ply`: https://github.com/TheBlackPlague/StockDory/blob/9f95dea6f73a5e790d33c5489fa1266dbf5e03d7/src/Engine/Search.h#L669

Let me know your thoughts.

Reply:
I prefer the ss +- ply approach, but this should be run through a poll once probably.